pkg/query-service/app/integrations/messagingQueues/kafka/sql.go

```diff
@@ -0,0 +1,171 @@
+package kafka
+
+import (
+	"fmt"
+)
+
+func generateConsumerSQL(start, end int64, topic, partition, consumerGroup, queueType string) string {
+	query := fmt.Sprintf(`
+WITH 
+-- Sub query for p99 calculation
+p99_query AS (
+    SELECT
+        serviceName,
+        quantile(0.99)(durationNano) / 1000000 as p99
+    FROM signoz_traces.signoz_index_v2
```

Comment:
You should be using `distributed_signoz_index_v2` everywhere otherwise it will only work with the local shard data query-service is connected to.

Reply:
done